core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3421,9 +3456,11 @@ class KafkaApis(val requestChannel: RequestChannel,
                              sharePartitionManagerInstance: SharePartitionManager,
                              authorizedTopics: Set[String],
                              groupId: String,
-                             memberId: String): CompletableFuture[Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData]] = {
+                             memberId: String,
+                             version: Short,
```

Comment:
I would tend to define this in a more abstract way such as `supportsRenewAcknowledgements : Boolean`. It is coincidence that both ShareFetch and ShareAcknowledge v2 support renew acks. The RPCs will diverge and I think it would be better not to push the actual versions down into the validation code.